src/net_processing.cpp

```diff
@@ -2539,6 +2547,28 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                       pfrom.nVersion.load(), peer->m_starting_height,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToString()) : ""),
                       pfrom.ConnectionTypeAsString());
+
+            // Keep track of the number of outbound flooding peers.
+            bool we_may_flood_to = false;
+            if (pfrom.MightSupportTransactionRelay()) {
+                if (m_reconciliation.IsPeerRegistered(pfrom.GetId())) {
+                    if (*m_reconciliation.IsPeerChosenForFlooding(pfrom.GetId())) {
```

Comment:
99906e0c3399225ee361ce85291074971d2025e1

`IsPeerRegistered` is not needed since `IsPeerChosenForFlooding` returns `nullopt`. Maybe:
```cpp
if (pfrom.MightSupportTransactionRelay()) {
    we_may_flood_to = m_reconciliation.IsPeerChosenForFlooding(pfrom.GetId()).value_or(false);
}
```

Reply:
I agree this is way cleaner, but I think the current code is easier to understand, so I'm leaning towards my approach.

Reply:
The problem is that you can hit undefined behavior with the current code.